### PR TITLE
Add whoami method to Go SDK

### LINKED ISSUE #760: Add whoami method to Go SDK
Sometimes, we need to simply check the validity of a user’s token

### PR DESCRIPTION
**Pull request by wilwell**
Add method `whoami` added to the Go SDK functionality. 

To invoke the method user only need to create a client with correct credentials (Cookie or Token for example). Unlike other yt methods, `whoami `response is JSON not YSON, and uses another handler path: `/auth/` instead of `api/v4`.

Right now the *testing* docker container of YT doesn't support the authentication, so for every request `whoami` we are getting `root` response. When the testing docker container will be fixed, the auth test also could be fixed.

Closes #760

### CODE REVIEW

**Comment by wilwell** [comment]
cc @l0kix2

File: yt/go/yt/integration/auth_client_test.go [hunk 1]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"crypto/sha256"
 	"fmt"
+	"go.ytsaurus.tech/yt/go/yt/ythttp"
+	"os"
```
Comment (l0kix2) [inline yt/go/yt/integration/auth_client_test.go:8]: It should be linters' job to say that, but these imports are not sorted properly. I suggest you configure your IDE for auto sorting (and we also need lint workflow here of course)

File: yt/go/yt/integration/auth_client_test.go [hunk 2]
```diff
@@ -65,6 +68,22 @@ func (s *Suite) TestIssueListRevokeToken(ctx context.Context, t *testing.T, yc y
...
 }
 
+func (s *Suite) TestWhoAmI(ctx context.Context, t *testing.T, yc yt.Client) {
+	user := "user-" + guid.New().String()
+	_ = s.CreateUser(ctx, t, user)
+	token, err := yc.IssueToken(ctx, user, "", nil)
+	require.NoError(t, err)
+	require.NotEmpty(t, token)
+
+	userClient, err := ythttp.NewClient(&yt.Config{Proxy: os.Getenv("YT_PROXY"), Logger: s.Env.L, Token: token})
```
Comment (l0kix2) [inline yt/go/yt/integration/auth_client_test.go:78]: Offtop, but I don't like this idea of passing proxy for test in env var, but I see that the current test run configuration doesn't have another way for it.

File: yt/go/yt/internal/call.go [hunk 3]
```diff
@@ -56,6 +58,11 @@ func (res *CallResult) decode(value any) (err error) {
 	return
 }
 
+func (res *CallResult) decodeJSON(value any) (err error) {
+	err = json.Unmarshal(res.YSONValue, value)
+	return
+}
```
Comment (l0kix2) [inline yt/go/yt/internal/call.go:64]: Could you provide some context (or maybe in the code comment also) 
1) why CallResult require JSON decoder now and didn't need it before? All other methods are returning YSON?
2)Am I understanding correctly that something called YSONValue contains JSON?

**Review by l0kix2 — CHANGES_REQUESTED** [review]

File: yt/go/yt/integration/auth_client_test.go (hunk 2 above)
Comment (wilwell) [reply yt/go/yt/integration/auth_client_test.go:78]: Yes, I agree that tis looks ugly, but I copied this method from the client creation in other testing methods.

File: yt/go/yt/internal/call.go (hunk 3 above)
Comment (wilwell) [reply yt/go/yt/internal/call.go:65]: 1. Even if we specify the output format is specified in header as YSON the whoami handler returns the JSON value.
2. YSONValue contains the byte representation of response body, it's inly named YSON

I've added couple more comments here and there.

**Review by l0kix2 — APPROVED** [review]
LGTM

File: yt/go/yt/internal/call.go [hunk 4]
```diff
@@ -56,6 +58,12 @@ func (res *CallResult) decode(value any) (err error) {
 	return
 }
 
+func (res *CallResult) decodeJSON(value any) (err error) {
+	// YSONValue contains the body of the response, it's not necessary YSON format, e.g. for whoami request
```
Comment (tinarsky) [inline yt/go/yt/internal/call.go:62]: nit: dot at the end of the comment.

**Review by tinarsky — CHANGES_REQUESTED** [review]

File: yt/go/yt/internal/call.go (hunk 4 above)
Comment (wilwell) [reply yt/go/yt/internal/call.go]: Done.

**Review by tinarsky — APPROVED** [review]
LGTM

**Pull request closed without merging**